CRM/Contact/Form/Task/PDFLetterCommon.php

```diff
@@ -136,11 +138,8 @@ public static function postProcess(&$form) {
       if (empty($caseId) && !empty($form->_caseIds[$item])) {
         $caseId = $form->_caseIds[$item];
       }
-      if ($caseId) {
-        $params['case_id'] = $caseId;
-      }
 
-      list($contact) = CRM_Utils_Token::getTokenDetails($params,
+      [$contact] = CRM_Utils_Token::getTokenDetails($params,
```

Comment:
getTokenDetails calls hook tokenValues and passes it params, so it would no longer contain the case_id.

Reply:
argh - so an entity model got hacked onto our call to retrieve contact tokens! ug.

I guess we'll just have to butcher up the case token processor to do something like

https://github.com/civicrm/civicrm-core/pull/21449

probably if it runs first it can 'get in first' - and we look to get some deprecation noise going

Fun fact - no test cover of it!

Reply:
Also I'm noticing contact custom fields aren't working in this context but not sure if it's from the PR or if it might have got borked by one of the previous ones.

Reply:
hmm we have test cover for contact fields for contact & for case - but perhaps not both in the same run.

Reply:
Sorry ignore the custom field part - my mistake.

Reply:
phew

Reply:
@demeritcowboy just looking at this - the contact returned in that row is already not used in handling the case tokens so this should be no-change on that front?

Reply:
What I'm thinking is the hook might use case_id to do something that alters the value that gets returned, which is then used in tplParams.

Reply:
@demeritcowboy yeah - it's hard to know what edge cases might be in play.

That `tplParams` assignment is a bit edge in itself because only sites that have smarty enabled for CiviMAIL via a define would even process that `$contact` - so we are looking at the risk that someone has a custom hook in play and that define in a magic combo.

I'm thinking that this PR is a no change on that front but we could

1) once #21449 is merged update the call to load tokens in there to pass 'case_id' if it is in the schema - per this
2) look at fixing tokenCompatsubscriber so that if smarty is enabled it assigns 'whatever is in contact token context at that point' to the template

If you agree I'll log gitlabs for those 2

Reply:
ok

Reply:
@demeritcowboy I found an existing issue - but what it basically says is 'the way it's currently doing it is wrong' & we should not do that.

I'm thinking that this was something added by Compucorp & it's not clear how deliberate it was. So my new proposal is not to add any handling to tokenCompatSubscriber but to create an extension (if there is demand) that could be installed if need be to call the hooks in the old way and do the contact to tpl assignment (that would mean we could rip out more of the cruft)

https://lab.civicrm.org/dev/core/-/issues/2554

Reply:
I've added https://lab.civicrm.org/dev/core/-/issues/2837 for the new one & using https://lab.civicrm.org/dev/core/-/issues/2554 for the existing one